prolog/regex.pl

```diff
@@ -40,8 +40,8 @@
     ).
 
 % macro expansion giving access to in-scope variables.
-user:goal_expansion(Text =~ Pattern, Goal) :-
-    % is goal expansion wanted?
+user:term_expansion(Text =~ Pattern, Goal) :-
+    % is term expansion wanted?
     prolog_load_context(module, Module),
```

Comment:
When running with this change, I get warnings like the following and 5 tests in t/captures.pl fail:

```
Warning: /home/mndrix/src/pack-regex/t/examples.pl:10:
        Local definition of user:regex/4 overrides weak import from regex
Warning: /home/mndrix/src/pack-regex/t/examples.pl:32:
        Clauses of regex/4 are not together in the source-file
          Earlier definition at /home/mndrix/src/pack-regex/t/examples.pl:10
          Current predicate: 'i beg with ^'/0
          Use :- discontiguous regex/4. to suppress this message
```

In case it helps, I have the following two files in my working directory that I use for testing:

```
$ cat dev.pl
:- assertz(file_search_path(library, '/home/mndrix/src/pack-regex/prolog')).
```

and

```
$ cat Makefile 
test:
        prove -e 'swipl -f dev.pl -q -t main -s' t/*
```